src/shop.c

```diff
@@ -1162,7 +1162,9 @@ static void Task_ReturnToItemListAfterItemPurchase(u8 taskId)
         PlaySE(SE_SELECT);
 
         // Purchasing 10+ Poke Balls gets the player a Premier Ball
-        if (tItemId == ITEM_POKE_BALL && tItemCount >= 10 && AddBagItem(ITEM_PREMIER_BALL, 1) == TRUE)
+        if (I_PREMIER_BALL_BONUS >= GEN_7 && (ItemId_GetPocket(tItemId) == POCKET_POKE_BALLS) && tItemCount >= 10 && AddBagItem(ITEM_PREMIER_BALL, tItemCount / 10) == TRUE)
+            BuyMenuDisplayMessage(taskId, (tItemCount >= 20 ? gText_ThrowInPremierBalls : gText_ThrowInPremierBall), BuyMenuReturnToItemList);
```

Comment:
Rather than just aborting giving 20 Premier Balls if you can only carry 19 more, shouldn't we just give the highest amount possible?

Reply:
I think I need to create a method for getting total amount of free space for an item.

Reply:
Actually, do we even know what happens in modern games? According to the vanilla Gen 3 code, it won't give you the Premier Ball if there's no space.

Let me boot Scarlet.

Reply:
Ok, here are some notes:
- The amount of Premier Balls received is shown in the message.
- If your purchase surpasses the amount of Premier Balls, you get maxed out.
- The message shows up how many Premier Balls you *would've gotten*, not how many you gained.
- The message still shows up if you buy at 999 Premier Balls.